com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/ProtoBuilder.java

```diff
@@ -44,6 +47,40 @@ public ProtoBuilder() {
         extToMessageClass.put(builderParams.outExt(), protoParams.messageClass());
     }
 
+    public static void addProtoDigest(Class<? extends GeneratedMessageV3> klass) throws NoSuchAlgorithmException {
+        if (classToProtoDigest.get(klass) == null) {
+            MessageDigest digest = MessageDigest.getInstance("SHA1");
+            digest.update(klass.getName().getBytes());
+            try {
+                // Calculate the digest for the proto format based on the proto descriptor
+                Descriptors.Descriptor descriptor = (Descriptors.Descriptor) klass.getMethod("getDescriptor").invoke(null);
+                digest.update(descriptor.getFullName().getBytes());
+                addFieldsToDigest(descriptor, digest);
+            } catch (ReflectiveOperationException e) {
+                throw new RuntimeException("Failed to retrieve descriptor from protobuf class", e);
+            }
+            classToProtoDigest.put(klass, digest.digest());
+        }
+    }
+
+    private static void addFieldsToDigest(Descriptors.Descriptor descriptor, MessageDigest digest) {
+        for (Descriptors.FieldDescriptor field : descriptor.getFields()) {
+            digest.update(field.getName().getBytes());
+            digest.update(field.getType().toString().getBytes());
+            digest.update(Integer.toString(field.getNumber()).getBytes());
+            if (field.hasDefaultValue()) {
+                digest.update(field.getDefaultValue().toString().getBytes());
```

Comment:
The digest includes field name, type, number and default value, so any change in proto file will change this hash digest